### PR TITLE
Tx detail layout

### PR DESCRIPTION
**Pull request by thatStrangeGuyThorchain**
Added UI components and stories for them:
- `TxType`
- `TxDetail`
as a part of #1017

### CODE REVIEW

File: src/renderer/services/midgard/types.ts [hunk 1]
```diff
@@ -154,3 +154,12 @@ export type PoolSharesLD = LiveData<Error, PoolShares>
...
 export type ValidateNodeLD = LiveData<ApiError, boolean>
+
+export type HistoryActionsPage = {
+  total: number
+  actions: HistoryActions
+}
+
+export type HistoryActionsPageRD = RD.RemoteData<ApiError, HistoryActionsPage>
+
+export type HistoryActionsPageLD = LiveData<ApiError, HistoryActionsPage>
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/midgard/types.ts:165]: needed for next PR. forgot to remove from this one. pls do not pay attention here

File: src/renderer/components/uielements/txType/TxType.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+const getIcon = (type: TxTypes) => {
+  switch (type) {
+    case TxTypes.DEPOSIT:
+      return <DepositIcon />
+    case TxTypes.WITHDRAW:
+      return <WithdrawIcon />
+    case TxTypes.SWAP:
+    case TxTypes.DOUBLE_SWAP:
+    default:
```
Comment (veado) [inline src/renderer/components/uielements/txType/TxType.tsx:25]:
```
default: return <></> // (no icon)
```

File: src/renderer/components/uielements/txType/TxType.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+const getIcon = (type: TxTypes) => {
+  switch (type) {
+    case TxTypes.DEPOSIT:
+      return <DepositIcon />
+    case TxTypes.WITHDRAW:
+      return <WithdrawIcon />
+    case TxTypes.SWAP:
+    case TxTypes.DOUBLE_SWAP:
```
Comment (veado) [inline src/renderer/components/uielements/txType/TxType.tsx:24]:
```
return <SwapIcon />
```

File: src/renderer/components/uielements/txType/TxType.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    case TxTypes.DEPOSIT:
+      return 'common.tx.type.deposit'
+    case TxTypes.WITHDRAW:
+      return 'common.tx.type.withdraw'
+    case TxTypes.SWAP:
+      return 'common.tx.type.swap'
+    case TxTypes.DOUBLE_SWAP:
+      return 'common.tx.type.doubleSwap'
+    default:
+      return 'common.tx.type.swap'
```
Comment (veado) [inline src/renderer/components/uielements/txType/TxType.tsx:41]: return '' as default

File: src/renderer/i18n/de/common.ts [hunk 5]
```diff
@@ -60,7 +60,11 @@ const common: CommonMessages = {
   'common.value': 'Wert',
   'common.modal.confirmTitle': 'Bestätige die Aktion',
   'common.manage': 'Manage',
-  'common.asset.base': 'Base'
+  'common.asset.base': 'Base',
+  'common.tx.type.swap': 'Swap - DE',
+  'common.tx.type.doubleSwap': 'Double swap - DE',
+  'common.tx.type.deposit': 'Deposit - DE',
+  'common.tx.type.withdraw': 'Withdraw - DE'
```
Comment (veado) [inline src/renderer/i18n/de/common.ts:67]: 'common.tx.type.swap': 'Swap', // as same as `common.swap`
  'common.tx.type.doubleSwap': 'Doppel Swap',
  'common.tx.type.deposit': 'Einzahlen', // similar to `common.add`
  'common.tx.type.withdraw': 'Auszahlen'

File: src/renderer/types/asgardex.ts [hunk 6]
```diff
@@ -62,3 +63,28 @@ export type TxStatus = {
...
+   * Inbound transactions related to the action
+   */
+  in: Tx[]
+  out: Tx[]
+  type: TxTypes
+  fees?: AssetWithAmount[]
+  slip?: number
+}
+
+export type HistoryActions = HistoryAction[]
```
Comment (veado) [inline src/renderer/types/asgardex.ts:90]: Let's move these types into `services/midgard/types.ts` or into `types/history`

**Review by veado — CHANGES_REQUESTED** [review]
LGTM! Just few suggestions...

File: src/renderer/components/uielements/txDetail/TxDetail.styles.ts [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    border-top-right-radius: 1.7rem;
+    border-bottom-right-radius: 1.7rem;
+    margin-bottom: 0;
+  }
+`
+
+export const InOutValue = styled(ContainerWithDelimeter)``
+
+export const InOutText = styled.span`
+  font-size: 0.75rem;
```
[the remark this answers is not in the document]
Comment (thatStrangeGuyThorchain) [reply src/renderer/components/uielements/txDetail/TxDetail.styles.ts:97]: hmmm. just an old habit 😅 _will change to pixels_

**Review by veado — APPROVED** [review]

**Pull request merged**